app/src/main/java/org/mozilla/fenix/home/blocklist/BlocklistMiddleware.kt

```diff
@@ -102,7 +118,18 @@ class BlocklistMiddleware(
                 topSites = topSites,
                 mode = mode,
                 collections = collections,
-                showCollectionPlaceholder = showCollectionPlaceholder
+                showCollectionPlaceholder = showCollectionPlaceholder,
+                recentSyncedTabState = when (recentSyncedTabState) {
+                    is RecentSyncedTabState.Success -> {
+                        val filteredTabs = recentSyncedTabState.tabs.filteredByBlocklist()
+                        if (filteredTabs.isEmpty()) {
+                            RecentSyncedTabState.None
+                        } else {
+                            RecentSyncedTabState.Success(filteredTabs)
+                        }
+                    }
+                    else -> recentSyncedTabState
+                }
```

Comment:
It looks like the logic here could potentially be re-used in the `AppAction.Change` block as well. I think it would be optimal for readability if it were possible to wrap this in an extension function that follows the `filteredByBlocklist` convention. Are there cases that we wouldn't want to switch the `RecentSyncedTabState` to `None` when there were no tabs after filtering?